posthog/temporal/common/posthog_client.py

```diff
@@ -12,6 +13,21 @@
     WorkflowInterceptorClassInput,
 )
 
+logger = structlog.get_logger()
+
+
+async def _add_inputs_to_properties(
+    properties: dict[str, Any],
+    input: ExecuteActivityInput | ExecuteWorkflowInput,
+    execution_type: Literal["activity", "workflow"],
+):
+    if len(input.args) == 1 and is_dataclass(input.args[0]) and hasattr(input.args[0], "properties_to_log"):
+        try:
+            properties.update(input.args[0].properties_to_log)
+        except Exception as e:
+            await logger.awarning("Failed to get safe properties for %s", input.args[0], exc_info=e)
```

Comment:
nit: Since now we have a notion of "safe" properties, we probably do not want to log the entire dataclass here, as dataclasses by default have a `__repr__` that logs everything, both unsafe and safe properties (which will be called by `__str__` by default). Ideally, we should override the dunder method and only print out "safe" properties.

Reply:
hmm yeah good point! I could probably just log the class name here to make things simpler